extensions/kubernetes-client/deployment/src/main/java/io/quarkus/kubernetes/client/deployment/KubernetesClientProcessor.java

```diff
@@ -20,14 +25,30 @@
     BuildProducer<ReflectiveClassBuildItem> reflectiveClasses;
 
     @BuildStep
-    public void process(CombinedIndexBuildItem combinedIndexBuildItem,
+    public void process(ApplicationIndexBuildItem applicationIndex, CombinedIndexBuildItem combinedIndexBuildItem,
             BuildProducer<ExtensionSslNativeSupportBuildItem> sslNativeSupport) {
         featureProducer.produce(new FeatureBuildItem(FeatureBuildItem.KUBERNETES_CLIENT));
 
+        Set<String> watchedClasses = new HashSet<>();
+        // make sure the watchers fully (and not weakly) register Kubernetes classes for reflection
+        applicationIndex.getIndex().getAllKnownImplementors(DotName.createSimple("io.fabric8.kubernetes.client.Watcher"))
+                .forEach(c -> {
+                    final MethodInfo eventReceived = c.firstMethod("eventReceived");
+                    if (eventReceived != null) {
```

Comment:
I think this is a bit too fragile and it would be better to rely on the type parameter of the `Watcher`. I'm not sure if we have a way to properly resolve it with Jandex though.

I need something similar for HV so it won't be a lost cause.

Reply:
Good point. Let me check and see if there is a way to do that

Reply:
Note that you would have to support something like:
```
WatcherImpl implements MyWatcher<Foo>
```
with
```
MyWatcher<S> extends Watcher<S>
```
We use classmate to do that in HV itself but having a pure Jandex solution would be handy.

We should share it somewhere and polish it over the time.

Reply:
At this point I would say I am comfortable enough with Jandex to make any changes to it. I was basically thinking of checking to see if it supports what is needed in this case and if so using it.
Does that make sense?

Reply:
You are or you are **not**?

I wasn't thinking about making changes to Jandex for now. I was wondering if we could write some code to use the information we have in the index to get the proper type parameter information expanded.

We could use it here, I think I would need it too for my latest changes to HV (maybe not the same exactly as I have 2 type parameters but something very similar).

I'm not sure how much work it would be though.

Reply:
Sorry, I meant I am **not** :P.

I see what you are saying and it makes perfect sense. I'll see what I can come up with :)

Reply:
Addressed